guides/source/active_record_multiple_databases.md

```diff
@@ -0,0 +1,258 @@
+end
+```
+
+By default Rails expects the database roles to be `writing` and `reading` for the primary
+and replica respectively. If you have a legacy system you may already have roles set up that
+you don't want to change. In that case you can set a new role name in your application config.
+
+```ruby
+config.active_record.writing_role = :default
+config.active_record.reading_role = :readonly
+```
+
+Now that we have the database.yml and the new model set up it's time to create the databases.
+Rails 6.0 ships with all the rails tasks you need to use multiple databases in Rails.
+
+You can run `rails -T` to see all the commands you're able to run. You should see the following:
+
+```
+$ rails -T
+rails db:create                          # Creates the database from DATABASE_URL or config/database.yml for the ...
+rails db:create:animals                  # Create animals database for current environment
+rails db:create:primary                  # Create primary database for current environment
+rails db:drop                            # Drops the database from DATABASE_URL or config/database.yml for the cu...
+rails db:drop:animals                    # Drop animals database for current environment
+rails db:drop:primary                    # Drop primary database for current environment
+rails db:migrate                         # Migrate the database (options: VERSION=x, VERBOSE=false, SCOPE=blog)
+rails db:migrate:animals                 # Migrate animals database for current environment
+rails db:migrate:primary                 # Migrate primary database for current environment
+rails db:migrate:status                  # Display status of migrations
+rails db:migrate:status:animals          # Display status of migrations for animals database
+rails db:migrate:status:primary          # Display status of migrations for primary database
+```
+
+Running a command like `rails db:create` will create both the primary and animals databases.
+Note that there is no command for creating the users and you'll need to do that manually
+to support the readonly users for your replicas. If you want to create just the animals
+database you can run `rails db:create:animals`.
+
+## Migrations
+
+Migrations for multiple databases should live in their own folders prefixed with the
+name of the database key in the configuration.
+
+You also need to set the `migrations_paths` in the database configurations to tell Rails
+where to find the migrations.
+
+For example the `animals` database would look in the `db/animals_migrate` directory and
+`primary` would look in `db/migrate`. Rails generators now take a `--database` option
+so that the file is generated in the correct directory. The command can be run like so:
+
+```
+$ rails g migration CreateDogs name:string --database animals
+```
+
+## Activating automatic connection switching
+
+Finally, in order to use multiple databases in your application you'll need to activate
+the middleware for automatic switching.
```

Comment:
"in order to use the read-only replica", maybe?

"to use multiple databases" sounds a bit too generic: that would equally describe e.g. a primary + animals setup, which doesn't need any switching.

Reply:
Good call, fixed.